src/runtime/cache.cpp

```diff
@@ -122,6 +120,15 @@ struct CacheEntry {
 
 };
 
+struct CacheBlockHeader {
+    uint32_t hash;
```

Comment:
Switch the order of these struct fields so that memory layout is clear. We've had problems in the past relating to this, so we've gotten into the habit of sorting struct fields in order of decreasing size.